### DISCUSSION TITLE
502 Bad Gateway

### DISCUSSION THREAD

**Discussion by eriksandall** [docksal/docksal#1759] [Q&A] [answered]
I consistently get a 502 Bad Gateway error when visiting my site (local dev). The full error that displays is: "502 Bad Gateway
openresty/1.19.9.1"

This site was created using the Drupal 7 boilerplate. 

1. Start computer
2. Start docker (I typically use Colima, but this error also occurs when using Docker Desktop)
3. run `fin project start`
4. Go to http://mysite.docksal.site 

I've tried running 'fin logs -f' before doing step 4, but no errors are recorded. 'docker ps' shows that all the necessary containers are running. 

I've tried restarting fin and docker, but that doesn't help. The only thing that works is stopping fin and docker, restarting the computer, then starting docker and fin. Sometimes it takes several restarts to work.

How else can I troubleshoot this?

Thanks in advance.

**UPDATE**: This might be an issue with Docker contexts. Running `docker context list` showed that the context was on 'default'. If I switch to 'docker-linux' (`docker context user docker-linux`), then start the container via Docker Desktop, then run `fin start`... I'm able to access my website. Seems like the context changes when I use Colima vs. Docker Desktop, and if the context isn't the correct one, then I'll get the 502 error. Does that make sense?

**Comment by eriksandall** [comment]
My update claiming that the issue was with Docker contexts was wrong. After setting the correct context, I continue to consistently experience this issue. Even after a fresh site install using the Docksal Drupal 7 boilerplate. Restarting docksal and Docker Desktop doesn't resolve the issue -- I actually have to log out of macOS, then log back in and start Docker and Docksal. How can I fix this? Everyday I waste about 15-20 minutes trying to get my dev environment up and running so I can actually work.

**Comment by lmakarov** [comment, marked as answer]
There is a fix in develop (#1764) that may address this issue. Please try `DOCKSAL_VERSION=develop fin update`

**Comment by eriksandall** [comment]
Thanks @lmakarov. After 2 weeks, this fix appears to be working to resolve the issue.
